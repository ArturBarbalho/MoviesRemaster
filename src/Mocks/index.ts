import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }