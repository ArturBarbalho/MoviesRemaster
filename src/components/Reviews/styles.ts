import styled from 'styled-components'
import media from 'styled-media-query'
export const Wrapper = styled.main``
export const Unit = styled.div`
display:flex;
margin-top:20px;
justify-content:center;
`
export const Img = styled.img`
width:50px;
border-radius:100px;
height:50px;
`
export const Right = styled.div`
max-width:1500px;
width:100vw;
margin-left:20px;
color:white;
h2{
    color:darkorange;
    margin-bottom:20px;
    }
`