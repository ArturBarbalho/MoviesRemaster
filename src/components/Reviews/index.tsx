import { reviewsProps } from '../../typesProps'
import * as S from './styles'
type props = {
  props:reviewsProps[]
}
const Reviews = ({props}:props) => (
  <S.Wrapper>
    {props.map((e,i)=>{
      return(
        <S.Unit key={i}>
        <S.Img  onError={({ currentTarget }) => {
        currentTarget.src='/user.png'}} src={e.src}/> 
        <S.Right>
        <h2>{e.name}</h2>
        <p>{e.content}</p>
      </S.Right>
      </S.Unit>
      )
    })}
  </S.Wrapper>
)

export default Reviews