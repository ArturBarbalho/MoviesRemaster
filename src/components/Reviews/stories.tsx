import { Story, Meta } from '@storybook/react/types-6-0'
import Reviews from '.'
import { ReviewsMock } from '../../Mocks'

export default {
  title: 'Reviews',
  component: Reviews
} as Meta
const mock = [ReviewsMock,ReviewsMock,ReviewsMock,ReviewsMock]
export const Default: Story = () => <Reviews props={mock} />