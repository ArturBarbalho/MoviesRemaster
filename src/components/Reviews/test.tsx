import { render, screen } from '@testing-library/react'
import Reviews from '.'
import { ReviewsMock } from '../../Mocks'

const mock = [ReviewsMock]

describe('<Reviews />', () => {
  it('should render the correct values', () => {
    const { container } = render(<Reviews props={mock} />)

    expect(screen.getByText('Mas devo explicar-lhe como nasceu toda essa id√©ia equivocada de denunciar um prazer e louvar a dor, e lhe darei um relato ...')).toBeInTheDocument()

    expect(screen.getByText('ONYANKOPON')).toBeInTheDocument()

    expect(screen.getByRole('img')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

})