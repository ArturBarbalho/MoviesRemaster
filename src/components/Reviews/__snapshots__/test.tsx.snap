// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews /> should render the correct values 1`] = `
<main
  class="styles__Wrapper-sc-1wkts46-0"
>
  <div
    class="styles__Unit-sc-1wkts46-1 IOSPO"
  >
    <img
      class="styles__Img-sc-1wkts46-2 kevLAL"
      src="https://image.tmdb.org/t/p/original//sKeC7qZLAKreuwxH4x6U3mN7Aa8.jpg"
    />
    <div
      class="styles__Right-sc-1wkts46-3 emiuqb"
    >
      <h2>
        ONYANKOPON
      </h2>
      <p>
        Mas devo explicar-lhe como nasceu toda essa id√©ia equivocada de denunciar um prazer e louvar a dor, e lhe darei um relato ...
      </p>
    </div>
  </div>
</main>
`;
